filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceDownloader.java

```diff
@@ -57,12 +57,15 @@ private void startDownload(Duration timeout, FileReferenceDownload fileReference
         }
         long end = System.currentTimeMillis() + timeout.toMillis();
         boolean downloadStarted = false;
+        int retryCount = 0;
         while ((System.currentTimeMillis() < end) && !downloadStarted) {
             try {
                 if (startDownloadRpc(fileReference)) {
                     downloadStarted = true;
                 } else {
-                    Thread.sleep(10);
+                    // Simple exponential backoff, might consider adding some randomness
+                    retryCount++;
+                    Thread.sleep(((int) Math.min(Math.round(Math.pow(2, retryCount)), 300) * 1000));
```

Comment:
Which thread pool is this.

Reply:
This is a thread pool in the FileDownloader (fixed, 10 threads) used by both config proxy and config server